fedimint-client/src/meta.rs

```diff
@@ -0,0 +1,375 @@
+                    revision: value.revision,
+                    value: parse_meta_value_static::<V>(&value.value).ok()?,
+                    priority: key.priority,
+                })
+            })
+    }
+
+    pub fn add_source(&mut self, source: impl MetaSource + 'static) {
+        self.sources.push(Box::new(source))
+    }
+
+    /// Update all source in background.
+    ///
+    /// Caller should run this method in a task.
+    pub(crate) async fn update_continuously(&self, client: &Client) -> ! {
+        // need to be fast the first time.
+        let metas = join_all(self.sources.iter().map(|source| async move {
+            (
+                source,
+                timeout(
+                    Duration::from_secs(60),
+                    retry(
+                        "update_source",
+                        FibonacciBackoff::default()
+                            .with_min_delay(Duration::from_millis(300))
+                            .with_max_delay(Duration::from_secs(3)),
+                        || source.fetch(client),
+                    ),
+                )
+                .await
+                .context("timeout while fetch source"),
+            )
+        }))
+        .await;
+        // initially write all once
+        let mut dbtx = client.db().begin_transaction().await;
+        for (source, meta) in metas {
+            if let Ok(meta) = meta.and_then(|x| x) {
+                Self::update_meta_values(&mut dbtx, source.priority(), source.source_id(), &meta)
+                    .await;
+            }
+        }
+        dbtx.commit_tx().await;
+        self.initial_fetch_waiter.done();
+
+        // now keep updating all
+        let values = join_all(
+            self.sources
+                .iter()
+                .map(|source| self.update_source_continuously(client, &**source)),
+        )
+        .await;
+        // to get ! type
+        values[0]
+    }
+
+    async fn update_source_continuously(&self, client: &Client, source: &dyn MetaSource) -> ! {
+        loop {
+            MetaSource::wait_for_update(source).await;
+            if let Ok(meta) = retry(
+                "update_source",
+                // very slow retry
+                FibonacciBackoff::default()
+                    .with_min_delay(Duration::from_secs(10))
+                    .with_max_delay(Duration::from_secs(10 * 60))
+                    .with_max_times(usize::MAX),
+                || source.fetch(client),
+            )
+            .await
+            {
+                let mut dbtx = client.db().begin_transaction().await;
+                Self::update_meta_values(&mut dbtx, source.priority(), source.source_id(), &meta)
+                    .await;
+                // TODO: what should have happen in case of conflicts?
+                // should we ignore the conflict or auto commit
+                let _ = dbtx.commit_tx_result().await;
+            }
+        }
+    }
+
+    async fn update_meta_values(
```

Comment:
`save_`? `store_`? "update" here is vague. "update" as in, fetch + store / fetch, etc.